StandaloneMmPkg/Drivers/StandaloneMmIplPei/MmFoundationHob.c

```diff
@@ -843,23 +916,33 @@ CreateMmFoundationHobList (
     UsedSize += HobLength;
   }
 
+  //
+  // Build resource HOB for unblocked region
+  //
   HobLength = GetRemainingHobSize (*FoundationHobSize, UsedSize);
-  if (PcdGetBool (PcdCpuSmmRestrictedMemoryAccess)) {
+  MmIplBuildResourceHobForUnblockedRegion (FoundationHobList + UsedSize, &HobLength);
+  UsedSize += HobLength;
+
+  if (!PcdGetBool (PcdCpuSmmRestrictedMemoryAccess)) {
+    UnblockRegion = NULL;
     //
-    // Only unblocked memory regions are accessible
+    // Collect unblocked memory regions
     //
-    MmIplBuildResourceHobForUnblockedRegion (FoundationHobList + UsedSize, &HobLength);
-  } else {
+    UnblockRegion = CollectUnblockedMemoryRegions (&UnblockRegionCount);

```

Comment:
The existing code pattern is CollectxxxRegions is called() from MmIplBuildResourceHobForAllSystemMemory.
Your code pattern is NOT like the existing one. Can you follow the exsiting pattern?

Reply:
Ray, existing code follow the code pattern that all excluded mem should be passed by the parameter

Reply:
As talked, moved CollectUnblockedMemoryRegions() to MmIplBuildResourceHobForAllSystemMemory()